Content.Client/Weapons/Melee/MeleeWeaponSystem.cs

```diff
@@ -115,21 +106,38 @@ public override void Update(float frameTime)
         {
             coordinates = EntityCoordinates.FromMap(MapManager.GetMapEntityId(mousePos.MapId), mousePos, TransformSystem, EntityManager);
         }
+        
+        // If the gun has AltFireComponent, it can be used to attack.
+        if (TryComp<GunComponent>(weaponUid, out var gun) && gun.UseKey)
+        {
+            if (!TryComp<AltFireMeleeComponent>(weaponUid, out var altFireComponent) || altDown != BoundKeyState.Down)
```

Comment:
A bit unsure about `gun.UseKey` and `altDown` usage here. If this is gated behind `UseKey`, is the `altDown` check even necessary? And does this mean that a gun that shoots via alt-use can't be made to do melee on Use?